.github/workflows/cd.yaml

```diff
@@ -0,0 +1,221 @@
+          PYPIPACKAGE=neuro-extras==$PYPIVERSION
+          until python -m pip install $PYPIPACKAGE
+          do
+              echo "Waiting for the pypi package $PYPIPACKAGE ..."
+              sleep 1
+          done
+      - name: Build release image
+        run: |
+          export PACKAGE="neuro-extras==${{ needs.info.outputs.version }}"
+          docker build -t neuromation/neuro-extras:latest \
+            --build-arg NEURO_EXTRAS_PACKAGE=$PACKAGE .
+      - name: Push release release image
+        id: push
+        run: |
+          export IMAGE=neuromation/neuro-extras
+          export TAG=${{ needs.info.outputs.version }}
+
+          echo "::set-output name=image::$IMAGE"
+          echo "::set-output name=tag::$TAG"
+
+          docker login $DOCKER_SERVER --username $DOCKER_USERNAME --password $DOCKER_PASSWORD
+          docker tag neuromation/neuro-extras:latest neuromation/neuro-extras:$TAG
+          docker push neuromation/neuro-extras:$TAG
+          # Push latest if it's not an alpha release:
+          #   not alpha: TAG="20.9.29"
+          #       alpha: TAG="20.9.29a1"
+          if [[ ! "$TAG" =~ ^.*a[0-9]*$ ]]; then
+            docker push neuromation/neuro-extras:latest
+          fi
+      - name: Wait until image becomes available on DockerHub
+        timeout-minutes: 10
+        run: |
+          export IMAGE=${{ steps.push.outputs.image }}
+          export TAG=${{ steps.push.outputs.tag }}
+
+          docker_tag_exists () {
+              CT="Content-Type: application/json"
+              U=$DOCKER_USERNAME
+              P=$DOCKER_PASSWORD
+              URL=https://hub.docker.com
+              TOKEN=$(curl -s -H "$CT" -X POST -d '{"username": "'$U'", "password": "'$P'"}' $URL/v2/users/login/ | jq -r .token)
+              curl --silent -f --head -lL $URL/v2/repositories/$1/tags/$2/ > /dev/null
+          }
+
+          until docker_tag_exists $IMAGE $TAG
+          do
+              echo "Waiting for the dockerhub image $IMAGE:$TAG ..."
+              sleep 1
+          done
+
+  test:
+    name: Run e2e tests against release image
+    needs: [dockerhub_deploy]
+    if: github.event_name == 'release'
+    strategy:
+      matrix:
+        python-version: [3.6, 3.7]
+        os: [ubuntu, macos, windows]
+      fail-fast: false
+    runs-on: ${{ matrix.os }}-latest
+    timeout-minutes: 30
+    env:
+      NEURO_STAGING_URL: ${{ secrets.NEURO_STAGING_URL }}
+      NEURO_TOKEN: ${{ secrets.NEURO_TOKEN }}
+      NEURO_CLUSTER: neuro-compute
+      DOCKER_SERVER: docker.io
+      DOCKER_USERNAME: ${{ secrets.DOCKER_USERNAME }}
+      DOCKER_PASSWORD: ${{ secrets.DOCKER_PASSWORD }}
+    steps:
+      - name: Checkout commit
+        uses: actions/checkout@v2
+      - name: Install python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Get pip cache dir
```

Comment:
Curiosity: why do we need it?

Reply:
well, it's used in the next step for pip caching. I copy-pasted this part from `ci.yml`